src/digraph.rs

```diff
@@ -1400,6 +1400,8 @@ impl PyDiGraph {
                 .graph
                 .neighbors(NodeIndex::new(node))
                 .map(|node| node.index())
+                .collect::<HashSet<usize>>()
+                .drain()
```

Comment:
One thing I was just wondering here is whether using a set here is faster or just sorting and calling dedup on the sorted vec is faster. something like:

```rust
let nodes: Vec<usize> =  self.graph.neighbors(NodeIndex::new(node)).map(|node| node.index()).collect();
nodes.sort_unstable();
nodes.dedup();
NodeIndices {
    nodes,
};
```
(we also have rayon's parallel sort functions too which can be faster depending on how large the input is, so probably not useful here)

Let me benchmark it quickly and see if there is a difference.

Reply:
I was looking at the `dedup()` option too. In the end, it was [this answer in Stack Overflow](https://stackoverflow.com/questions/47636618/vecdedup-does-not-work-how-do-i-deduplicate-a-vector-of-strings#answer-47648303) and the fact we were [already using it in the codebase for similar purposes](https://github.com/Qiskit/retworkx/blob/master/src/digraph.rs#L547) that pushed me in the HashSet direction.

Reply:
Yeah, that makes sense. It's also consistent with the quick benchmarks I ran to test it too:

```python
import retworkx
import time

graph = retworkx.directed_gnp_random_graph(1024, .95, seed=42)
graph.add_edges_from_no_data(list(graph.edge_list()))
start = time.time()
graph.neighbors(512)
stop = time.time()
print("rand time: %s" % str(stop - start))

star = retworkx.generators.directed_star_graph(1024)
star.add_edges_from_no_data(list(star.edge_list()))
start = time.time()
star.neighbors(0)
stop = time.time()
print("star time: %s" % str(stop - start))
```

which returned:
```
Master:
rand time: 3.0517578125e-05
star time: 5.7220458984375e-06

This PR:
rand time: 3.3855438232421875e-05
star time: 3.4332275390625e-05

Vec dedup:
rand time: 4.9114227294921875e-05
star time: 4.029273986816406e-05

Vec parallel sort dedup:
rand time: 4.029273986816406e-05
star time: 4.2438507080078125e-05
```

either way it's not going to huge difference, but this is definitely the better way to handle it for our use case.